test/cpp/jit/test_gpu_shift.cpp

```diff
@@ -2245,6 +2245,183 @@ TEST(NVFuserTest, FusionHdiff_CUDA) {
   }
 }
 
+TEST(NVFuserTest, FusionHdiffPartialSplit_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto inp = makeSymbolicTensor(3);
+  fusion.addInput(inp);
+  auto coeff = makeSymbolicTensor(3);
+  fusion.addInput(coeff);
+
+  std::vector<std::vector<int>> offsets{
+      {0, 1, 0}, {0, -1, 0}, {0, 0, 1}, {0, 0, -1}};
+
+  // T2, T3, T4, T5
+  std::vector<TensorView*> inp_neighbors;
+  for (const auto& offset : offsets) {
+    inp_neighbors.push_back(shift(inp, offset, false));
+  }
+
+  // T8
+  TensorView* sum_of_neighbors = nullptr;
+  for (auto inp_neighbor : inp_neighbors) {
+    if (sum_of_neighbors == nullptr) {
+      sum_of_neighbors = inp_neighbor;
+    } else {
+      sum_of_neighbors = add(sum_of_neighbors, inp_neighbor);
+    }
+  }
+
+  // T9 = T0 * 4
+  // T10 = T9 - T8
+  auto lap = sub(mul(inp, new Double(4)), sum_of_neighbors);
+
+  // T11 = shift(T10)
+  // T12 = T11 - T10
+  auto flx = sub(shift(lap, {0, 0, -1}, false), lap);
+  // T14 = T13 - T0
+  // T15 = T12 * T14
+  // T16 = T15 > 0
+  // T17 = T16 ? 0 : T12
+  auto flx_cond =
+      gt(mul(flx, sub(shift(inp, {0, 0, -1}, false), inp)), new Double(0));
+  auto flx0 = where(flx_cond, new Double(0), flx);
+
+  // T18 = shift(T10)
+  // T19 = T18 - T10
+  auto fly = sub(shift(lap, {0, -1, 0}, false), lap);
+  // T20 = shift(T0)
+  // T21 = T20 - T0
+  // T22 = T19 * T21
+  // T23 = T22 > 0
+  auto fly_cond =
+      gt(mul(fly, sub(shift(inp, {0, -1, 0}, false), inp)), new Double(0));
+  // T24 = T23 ? 0 : T19
+  auto fly0 = where(fly_cond, new Double(0), fly);
+
+  // T25 = shift(flx0)
+  // T26 = T17 - T25
+  // T27 = shift(fly0)
+  // T28 = T24 - T27
+  // T29 = T26 + T28
+  // T30 = T1 * T29
+  // T31 = T0 - T30
+  auto out =
+      sub(inp,
+          mul(coeff,
+              add(sub(flx0, shift(flx0, {0, 0, 1}, false)),
+                  sub(fly0, shift(fly0, {0, 1, 0}, false)))));
+
+  fusion.addOutput(out);
+
+  /////////////////////////////////
+  // Scheduling
+  /////////////////////////////////
+
+  // Step 1: 2D Tiling
+
+  const int tile_x = 32;
+  const int tile_y = 8;
+
+  out->split(-1, tile_x, true, true);
+  out->split(-3, tile_y, true, true);
+  out->reorder({{-2, -3}});
+  inp->computeAt(out, -3);
+  coeff->computeAt(out, -3);
+
+  // Step 2: Inlining
+
+  // Inline inputs to lap
+  auto lap_vals = DependencyCheck::getAllValsBetween({inp}, {lap});
```

Comment:
It's actually not possible yet as `MostInlined` does not know about constraints specific to shift. Specifically, it's not possible to inline an axis to another axis if the former axis has larger halo. For example:

```
auto tv0 = makeSymbolicTensor(1);
auto tv1= add(tv0, new Double(1));
auto avg = add(tv1, add(shift(tv1, {-1}), shift(tv1, {1})));
```

It's not possible to inline `tv1` at -1 position of `avg` since `avg` needs 3 values from `tv1` for each of `avg` elements. This constraint should be embedded in the logic to discover legal computeAt positions, but I haven't worked on that yet. Created an issue to work on it later (#1094)